mat_dp_pipeline/data_sources/iam.py

```diff
@@ -0,0 +1,100 @@
+import logging
+from pathlib import Path
+from typing import ClassVar, Final
+
+import pandas as pd
+
+from mat_dp_pipeline.abstract_data_sources import TargetsSource
+from mat_dp_pipeline.data_sources.country_sets.country_set import Identifier
+from mat_dp_pipeline.data_sources.country_sets.source_with_countries import (
+    SourceWithCountries,
+)
+
+from .common import map_technologies
+from .tech_map import TechMap, TechMapTypes, create_tech_map
+
+
+class IntegratedAssessmentModel(TargetsSource):
+    _spreadsheet_path: Path
+    _parameters: list[str]
+    _country_to_path: dict[str, Path]
+    _grouping: Final[tuple[str, ...]] = ("Region", "Model", "Scenario", "Parameter")
+    _tech_map: ClassVar[TechMap] = create_tech_map(TechMapTypes.IAM)
+    tail_labels: ClassVar[list[str]] = ["Model", "Scenario", "Parameter"]
+
+    def __init__(
+        self,
+        spreadsheet_path: Path,
+        parameters: list[str],
+        country_source: type[SourceWithCountries],
+    ) -> None:
+        """Targets Data Sources for Integrated Assement Model (TIAM).
+
+        The spreadsheet must have a tab called "DATA_TIAM".
+
+        Args:
+            spreadsheet_path (Path): path to the spreadsheet on the hard drive.
+            parameters (list[str]): list of parameters such as "Primary Energy", "Final Energy".
+                                    Essentially these are prefixes of values in `variable` column
+                                    of the spreadsheet.
+            country_source (type[SourceWithCountries]):
+                Data sources with countries. This normally would be the type of intensities source
+                used. This is required to guarantee correct country/region mapping to the paths between
+                the intensities/indicators and the targets.
+        """
+        assert len(parameters) > 0
```

Comment:
Could we not just say if len(parameters) == 0, we are interested in the entire file? No worries if this is complicated to implement